all_teachers.rb

```diff
@@ -0,0 +1,70 @@
+class All_Teachers
```

Comment:
Naming nitpicks: class names should be `CamelCase` -- no underscores.  Also, class names should be singular, and should describe what a single instance of that class would be.  An instance of this class would be... an `all_teachers` object?  I'd go with `Instructor` or even `GenericTeacher` here.  Remember, when you see `Teacher < All_Teachers`, that says "A `Teacher` is an `All_Teachers`".  If the grammar doesn't work, the name doesn't work. (Note that saying "A `Teacher` is an `Instructor`" does sound natural, at least to me.)